### ISSUE TITLE
UIPageableData: do not validate integers in isRowsPerPageValid

### ISSUE THREAD

**Issue by djmj** [primefaces/primefaces#8036]
Migrating to primefaces 10.0.0 there are some markups of datatable with paginator in our app like: `rows="5" rowsPerPageTemplate="10, 20, 50"`

Now this throws an exception now that 5 is not in the template which is not an error. Rows must not be in template, else template would be required so i think those options would be more developer friendly:

- dont check valid integer option like it worked before 
- silently add rows to template behind the back on init
- what about dynamic rows? `rows="#{bean.rows}"`

During migration i now have to check all tables where rows is missing in the template definition. It was a convenient shortcut before.

java.lang.IllegalArgumentException: Unsupported rows per page value: 5
	at org.primefaces.component.api.UIPageableData.updatePaginationData(UIPageableData.java:330)

**Idea**

https://github.com/primefaces/primefaces/blob/d8acc5196925c234c69c143036b02619a72d0eb9/src/main/java/org/primefaces/component/api/UIPageableData.java#L290 

If the `rowsParam` is parseable to an integer it should be valid and this could be checked before checking the template. This could be checked with try-parse method or by regex testing if its a number.

**Comment by tandraschko** [comment]
we changed this by design that "rows" must be inside the "template" because of security: https://github.com/primefaces/primefaces/issues/3519

so you need to add it to the template
dynamic rows doesnt matter if it doesnt change between rendering and the next DT request

**Comment by tandraschko** [comment]
and NOTE, that change was in PF 7 ;)

**Comment by djmj** [comment]
Makes sense. 

Yeah i was stuck since mobile was removed in 7 and had no time before to migrate legacy mobile to higher versions until recently: https://github.com/djmj/primefaces-mobile

Now we migrated directly from 6.1 to 10.0.0 and today was first production day after migration and we faced big OOM memory leaks till crash every 30 minutes. Not sure where it comes from (my mobile fork or PF 10) but had to go back to 6.1 again. We are doomed to stay there.
